src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -238,11 +223,26 @@ void PacketDataReporter::OnResource(ResourceType type, const ResourceData & data
         }
         else if (mDiscoveryType == DiscoveryType::kOperational)
         {
-            OnOperationalSrvRecord(data.GetName(), srv);
+            // Ensure this is our record.
+            if (HasQNamePart(data.GetName(), kOperationalServiceName))
+            {
+                OnOperationalSrvRecord(data.GetName(), srv);
+            }
+            else
+            {
+                mValid = false;
```

Comment:
This is marking the whole response as invalid even though the record we're looking for might be later in the packet, causing https://github.com/project-chip/connectedhomeip/issues/9099. In fact this is almost always happening in some test scenarios with thread devices.